src/activities/jumbled_words/jumbled_words.js

```diff
@@ -0,0 +1,196 @@
+.import "qrc:/gcompris/src/core/core.js" as Core
+.import "qrc:/gcompris/src/activities/lang/lang_api.js" as Lang
+
+
+var url = "qrc:/gcompris/src/activities/jumbled_words/resource/"
+var defaultLevelsFile = "qrc:/gcompris/src/activities/lang/resource/words.json";
+var baseUrl = "qrc:/gcompris/src/activities/lang/resource/";
+
+var currentLevel
+var maxLevel
+var maxSubLevel
+var currentSubLevel
+var levels
+var level
+var items
+var index
+var answers
+var jumble=[]
+var dataset
+var maxWordInLesson = 12
+var locale
+var flatWordList
+
+function start(items_) {
+
+    items = items_
+    loadLevels();
+    currentLevel = 0
+    currentSubLevel = 0
+    maxLevel = levels.length
+    initLevel()
+}
+
+function loadLevels()
+{
+    locale = GCompris.ApplicationInfo.getVoicesLocale(items.locale)
+
+    // register the voices for the locale
+    GCompris.DownloadManager.updateResource(GCompris.DownloadManager.getVoicesResourceForLocale(locale))
+
+    dataset = Lang.load(items.parser, baseUrl,
+                        GCompris.ApplicationSettings.wordset ? "words.json" : "words_sample.json",
+                        "content-"+ locale +".json")
+
+    levels = Lang.getAllLessons(dataset)
+}
+
+function stop() {
+}
+
+function shuffleString(s)
+{
+    var a = s.split("");
+    var n = a.length;
+
+    for(var i = n-1; i>0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
+        var tmp = a[i];
+        a[i] = a[j];
+        a[j] = tmp;
+    }
+    return a.join("");
+}
+
+function generateKeyboard(questions){
+    var letters = new Array();
+    items.keyboard.shiftKey = false;
+    var a = questions.split("");
+    var n = a.length;
+    for(var i = n-1; i>=0; i--)
+    {
+     letters.push(a[i]);
+    }
+    // generate layout from letter map
+    var layout = new Array();
+    var row = 0;
+    var offset = 0;
+    while (offset < letters.length-1) {
+        var cols = letters.length <= 10 ? letters.length : (Math.ceil((letters.length-offset) / (3 - row)));
+        layout[row] = new Array();
+        for (var j = 0; j < cols; j++)
+            layout[row][j] = { label: letters[j+offset] };
+        offset += j;
+        row++;
+    }
+    items.keyboard.layout = layout;
+}
+
+function initLevel()
+{
+    items.bar.level = currentLevel + 1
+    var questions
+    flatWordList = Lang.getLessonWords(dataset, levels[currentLevel]);
+    var i = 0
+
+    items.hintimage.source=Qt.binding(function() { return ""});
+    if (currentSubLevel === 0) {
+        maxSubLevel = flatWordList.length;
+        jumble=[];
+        for(i=0;i<maxSubLevel ; i++)
+        {
+          jumble.push(i);
+        }
+        jumble=Core.shuffle(jumble);
+        items.score.numberOfSubLevels = maxSubLevel;
+        items.score.currentSubLevel = 1;
+    }
+    else {
+        items.score.currentSubLevel = currentSubLevel + 1;
+    }
+    index=jumble[currentSubLevel];
+    answers=flatWordList[index].translatedTxt
+    do{
+        if(answers.indexOf(" "))
```

Comment:
Do you really want to check if there is a space at first character or if there are no spaces at all? If second, indexOf returns -1 in case the string you look for is not in the word